ngraph/frontend/paddlepaddle/src/op/lstm.cpp

```diff
@@ -0,0 +1,212 @@
+                                            1);
+
+                            // Get dimensions needed for default inputs creation
+                            // Parsing init hidden state
+                            auto shape_of_x = std::make_shared<opset6::ShapeOf>(input_x);
+
+                            auto axes =
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {0});
+
+                            auto batch_size_node = std::make_shared<opset6::Gather>(
+                                    shape_of_x,
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {0}),
+                                    axes);
+
+                            auto seq_length_node = std::make_shared<opset6::Gather>(
+                                    shape_of_x,
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {1}),
+                                    axes);
+
+                            auto shape_of_r = std::make_shared<opset6::ShapeOf>(
+                                    m_input_map[LSTMInput::LSTM_INPUT_R]);
+                            auto num_directions_node = std::make_shared<opset6::Gather>(
+                                    shape_of_r,
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {0}),
+                                    axes);
+                            auto hidden_size_node = std::make_shared<opset6::Gather>(
+                                    shape_of_r,
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {2}),
+                                    axes);
+
+                            m_input_map[LSTMInput::LSTM_INPUT_SEQ_LENGTHS] =
+                                    std::make_shared<opset6::Broadcast>(seq_length_node,
+                                                                        batch_size_node);
+
+                            auto init_states = node.get_ng_inputs("PreState");
+                            // 0 for init_h, 1 for init_cell, update bidirect_len for init states
+                            bidirect_len = node.get_attribute<bool>("is_bidirec") ? 2 : 1;
+
+                            auto h_begin = opset6::Constant::create(element::Type_t::i64, {1}, {layer * bidirect_len});
+                            auto h_end = opset6::Constant::create(element::Type_t::i32, Shape{1}, {layer * bidirect_len + bidirect_len});
+                            auto c_begin = opset6::Constant::create(element::Type_t::i64, {1}, {layer * bidirect_len});
+                            auto c_end = opset6::Constant::create(element::Type_t::i64, {1}, {layer * bidirect_len + bidirect_len});
+
+                            m_input_map[LSTMInput::LSTM_INPUT_INIT_H] = builder::opset1::reorder_axes(std::make_shared<opset6::StridedSlice>(init_states[0], h_begin, h_end, std::vector<int64_t>{0}, std::vector<int64_t>{0}), {1, 0, 2});
```

Comment:
`std::make_shared<opset6::StridedSlice>` is unnecessary,  LSTMSequence (https://docs.openvinotoolkit.org/latest/openvino_docs_ops_sequence_LSTMSequence_1.html) is already sequence of LSTM units (LSTMCells in OpenVINO notation, https://docs.openvinotoolkit.org/latest/openvino_docs_ops_sequence_LSTMCell_1.html)

Reply:
This is because that the PDPD saves its hidden states in a different way with OV. It compresses the hidden state of each layer in RNN ["PreState"] node. We need to extract the related hidden state for this layer.

Reply:
Thanks for the explanation! Resolved